service/service_construction.go

```diff
@@ -96,14 +102,22 @@ func (s ConstructionService) ConstructionMetadata(
 
 	var gasLimit uint64
 	if input.GasLimit == nil {
-		if input.Currency == nil || utils.Equal(input.Currency, mapper.AvaxCurrency) {
+		if input.Currency == nil || types.Hash(input.Currency) == types.Hash(mapper.AvaxCurrency) {
 			gasLimit, err = s.getNativeTransferGasLimit(ctx, input.To, input.From, input.Value)
+			if err != nil {
+				return nil, wrapError(errClientError, err)
+			}
 		} else {
-			gasLimit, err = s.getErc20TransferGasLimit(ctx, input.To, input.From, input.Value, input.Currency)
-		}
-
-		if err != nil {
-			return nil, wrapError(errClientError, err)
+			if input.Metadata != nil {
+				if input.Metadata.UnwrapBridgeTx {
+					gasLimit, err = s.getBridgeUnwrapTransferGasLimit(ctx, input.From, input.Value, input.Currency)
+				}
```

Comment:
We should probably error in the `else` case here? Then gasLimit will be 0?

Reply:
I think you may have meant to do: `input.Metadata != nil && input.Metadata.UnwrapBridgeTx`

Reply:
#188